src/ngraph/frontend/onnxifi/backend.cpp

```diff
@@ -0,0 +1,312 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <limits>
+
+#include <onnxifi.h>
+
+#include "backend.hpp"
+
+namespace ngraph
+{
+    namespace onnxifi
+    {
+        void Backend::get_capabilities(void* info_value, std::size_t* info_value_size) const
+        {
```

Comment:
Please add @TODO comments indicating what these methods already do and what else is left to do for full ONNXIFI compliance.